app/controllers/SurveyController.scala

```diff
@@ -0,0 +1,121 @@
+        val allSurveyQuestionIds = allSurveyQuestions.map(_.surveyQuestionId)
+        val answeredQuestions = submission.answeredQuestions
+        val answeredQuestionIds = submission.answeredQuestions.map(_.surveyQuestionId.toInt)
+        val unansweredQuestionIds = allSurveyQuestionIds diff answeredQuestionIds
+        // Iterate over all the questions and check if there is a submission attribute matching question id.
+        // Add the associated submission to the user_submission tables for that question
+
+
+        answeredQuestions.foreach{ q =>
+          val questionId = q.surveyQuestionId.toInt
+          val temp_question = SurveyQuestionTable.getQuestionById(questionId)
+          temp_question match{
+            case Some(question) =>
+              if (question.surveyInputType != "free-text-feedback") {
+                val userSurveyOptionSubmission = UserSurveyOptionSubmission(0, userId, question.surveyQuestionId, Some(q.answerText.toInt), timestamp, numMissionsCompleted)
+                val userSurveyOptionSubmissionId: Int = UserSurveyOptionSubmissionTable.save(userSurveyOptionSubmission)
+              }
+              else {
+                val userSurveyTextSubmission = UserSurveyTextSubmission(0, userId, question.surveyQuestionId, Some(q.answerText), timestamp, numMissionsCompleted)
+                val userSurveyTextSubmissionId: Int = UserSurveyTextSubmissionTable.save(userSurveyTextSubmission)
+              }
+            case None =>
+              None
+          }
+        }
+        unansweredQuestionIds.foreach{ questionId =>
+          val temp_question = SurveyQuestionTable.getQuestionById(questionId)
+          temp_question match{
+            case Some(question)=>
+              if(question.surveyInputType != "free-text-feedback"){
+                val userSurveyOptionSubmission = UserSurveyOptionSubmission(0, userId, question.surveyQuestionId, None, timestamp, numMissionsCompleted)
+                val userSurveyOptionSubmissionId: Int = UserSurveyOptionSubmissionTable.save(userSurveyOptionSubmission)
+              }
+              else{
+                val userSurveyTextSubmission = UserSurveyTextSubmission(0, userId, question.surveyQuestionId, None, timestamp, numMissionsCompleted)
+                val userSurveyTextSubmissionId: Int = UserSurveyTextSubmissionTable.save(userSurveyTextSubmission)
+              }
+            case None =>
+              None
+          }
+        }
+
+        Future.successful(Ok(Json.obj("survey_success" -> "True")))
+      }
+    )
+
+  }
+
+  def shouldDisplaySurvey = UserAwareAction.async { implicit request =>
+    val userId: UUID = request.identity match {
+      case Some(user) => user.userId
+      case None =>
+        val user: Option[DBUser] = UserTable.find("anonymous")
+        UUID.fromString(user.get.userId)
+    }
+    val userRoles = UserRoleTable.getRoles(userId)
+
+    val numMissionsBeforeSurvey = 2
+    val userRoleForSurvey = "User"
```

Comment:
Update to Turker

Reply:
Or allow both. We should discuss with team whether we want to enable the survey for our volunteers as well now or later. If not now, then just keep to Turker role.

Reply:
For now its just Turker.